### ISSUE TITLE
Segmentation fault in gtest_filter="SUMMA*" with 8 processes

### ISSUE THREAD

**Issue by tuxamito** [dash-project/dash#348]
On current development branch (tag dash-0.2.0-2506-g47b12a9)

System:
Linux Mint 18.1 (based on Ubuntu LTS 16.04)
4+HT Skylake (i7-6700K)
mpic++ gcc version 5.4.1 20160904
mpirun (Open MPI) 1.10.2

There is a segmentation fault when executing the dash-test-mpi application as follows:
`mpirun -n 8  dash-test-mpi --gtest_filter="SUMMA*"`
while with 4 processes it ends apparently correctly.

I attach the output with valgrind for -n4 and -n8 for this command:
`mpirun -n x valgrind ./dash-test-mpi --gtest_filter="SUMMA*"`

[output_N4.txt](https://github.com/dash-project/dash/files/872937/output_N4.txt)
[output_N8.txt](https://github.com/dash-project/dash/files/872938/output_N8.txt)

**Comment by fuchsto** [comment]
Good catch, thank you. I have an idea what might have caused this, will report back here.
